hera_pspec/pspecdata.py

```diff
@@ -1721,6 +1906,30 @@ def pspec(self, bls1, bls2, dsets, pols, n_dlys=None, input_data_weight='identit
                         if verbose: print("  Computing and multiplying scalar...")
                         pv *= scalar * self.scalar_delay_adjustment(key1, key2, sampling=sampling)
 
+                    #Generate the covariance matrix if error bars provided
+                    if covariance:
+                        if verbose: print(" Building q_hat covariance...")
+                        cov_qv=np.zeros((self.Ntimes,
+                                         self.spw_Ndlys,
+                                         self.spw_Ndlys),dtype=complex)
+                        cov_pv=np.zeros_like(cov_qv)
+                        amat,bmat=np.meshgrid(range(self.spw_Ndlys),
+                                              range(self.spw_Ndlys))
+                        amat=amat.astype(int)
+                        bmat=bmat.astype(int)
+                        for tnum in range(self.Ntimes):
+                            cov_qv[tnum,:,:]=np.vectorize(lambda a,b:\
+                            self.cov_q_hat(a,b,key1,key2,time_indices=[tnum]))\
+                            (amat,bmat)
+                        if verbose: print(" Building p_hat covariance...")
+                        for tnum in range(self.Ntimes):
+                            cov_pv[tnum]=self.cov_p_hat(Mv,cov_qv[tnum,:,:])
```

Comment:
Same sort of comment. Should we consider folding the time loops (this is starting to sound like science fiction) into the `cov_p_hat`? I guess the only thing we want to make sure of is that the special case of having one time is still supported, so that if someone wanted to loop over time explicitly somewhere else, they can still do that.